format-avro/src/main/java/io/cdap/plugin/format/avro/AvroToStructuredTransformer.java

```diff
@@ -91,9 +98,96 @@ public Schema convertSchema(org.apache.avro.Schema schema) throws IOException {
     if (schemaCache.containsKey(hashCode)) {
       structuredSchema = schemaCache.get(hashCode);
     } else {
-      structuredSchema = Schema.parseJson(schema.toString());
+      String strSchema = schema.toString();
+      String jsonSchema = preprocessSchema(GSON.fromJson(strSchema, JsonObject.class)).toString();
+      structuredSchema = Schema.parseJson(jsonSchema);
       schemaCache.put(hashCode, structuredSchema);
     }
     return structuredSchema;
   }
+
+  private JsonObject preprocessSchema(JsonObject schema) {
```

Comment:
In one place you are using returned JsonObject (that is always the input as of now), in others you assume it's mutating the input. Please use same style.

Reply:
I've changed it so that every method that accepts a `JsonObject` modifies it in place and returns nothing.